go.mod

```diff
@@ -37,51 +37,47 @@ require (
 	k8s.io/apiextensions-apiserver v0.21.1
 	k8s.io/apimachinery v0.21.1
 	k8s.io/apiserver v0.21.1
-	k8s.io/client-go v12.0.0+incompatible
+	k8s.io/client-go v0.21.1
 	k8s.io/cloud-provider v0.21.1
 	k8s.io/component-helpers v0.21.1
 	k8s.io/kube-controller-manager v0.21.1
-	k8s.io/kubernetes v1.21.1
-	k8s.io/utils v0.21.1
-	sigs.k8s.io/controller-runtime v0.9.0
-	sigs.k8s.io/kustomize/kyaml v0.10.17
+	k8s.io/utils v0.0.0-20210527160623-6fdb442a123b
+	sigs.k8s.io/controller-runtime v0.8.0
```

Comment:
#8087 just bumped the controller runtime to v0.9. Can you try with 0.9? Perhaps that is contributing to the CI issues with the build as well.

Reply:
Well, that's embarrassing... There was a backwards incompatible change to some k8s metrics package (I forget which one) and controller-runtime v0.8.0 hadn't jumped passed it yet, which caused all kinds of problems. The reason it took so long for me to update this PR is because I was waiting for v0.9.0, and then I messed up the rebase ;x